### PR TITLE
cmd/services: note when Homebrew is too old.

### PR DESCRIPTION
**Pull request by MikeMcQuaid**
This is the version required for `AbstractCommand` to work.

### CODE REVIEW

File: cmd/services.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 # frozen_string_literal: true
 
+if Version.new(HOMEBREW_VERSION) < Version.new("4.2.15")
```
Comment (laggardkernel) [inline cmd/services.rb:3]: `HOMEBREW_VERSION` may be non-numeric string, and break this comparision. From `brew.sh`:

```
HOMEBREW_VERSION="$("${HOMEBREW_GIT}" -C "${HOMEBREW_REPOSITORY}" describe --tags --dirty --abbrev=7 2>/dev/null)"
HOMEBREW_USER_AGENT_VERSION="${HOMEBREW_VERSION}"
if [[ -z "${HOMEBREW_VERSION}" ]]
then
  HOMEBREW_VERSION=">=4.1.0 (shallow or no git repository)"
  HOMEBREW_USER_AGENT_VERSION="4.X.Y"
fi
```

Comment (MikeMcQuaid) [reply cmd/services.rb:3]: @laggardkernel I've tested this and the shallow/X.Y cases still handle as expected and fail here. Are you seeing differently?

Comment (xel1045) [reply cmd/services.rb:3]: This new check breaks `brew services` when its run with `sudo`:

```
$ sudo brew services list    
Error: Your Homebrew is too outdated for `brew services`. Please run `brew update`!

$ brew --version
Homebrew 4.2.16

$ sudo brew --version
Homebrew >=4.1.0 (shallow or no git repository)
```

Comment (laggardkernel) [reply cmd/services.rb:3]: I got it. it only fails when running services with `sudo brew services`. 

```
❯ brew -v
Homebrew 4.2.16-48-gb23db35
Homebrew/homebrew-core (git revision e7d9b7c9f07; last commit 2024-04-02)
Homebrew/homebrew-cask (git revision fe196ff5a4; last commit 2024-04-02)

❯ sudo brew -v
Homebrew >=4.1.0 (shallow or no git repository)
fatal: detected dubious ownership in repository at '/opt/homebrew/Library/Taps/homebrew/homebrew-core'
To add an exception for this directory, call:

        git config --global --add safe.directory /opt/homebrew/Library/Taps/homebrew/homebrew-core
Homebrew/homebrew-core (no Git repository)
fatal: detected dubious ownership in repository at '/opt/homebrew/Library/Taps/homebrew/homebrew-cask'
To add an exception for this directory, call:

        git config --global --add safe.directory /opt/homebrew/Library/Taps/homebrew/homebrew-cask
Homebrew/homebrew-cask (no Git repository)
```

**Pull request merged**